src/lib.rs

```diff
@@ -237,9 +246,9 @@ pub fn derive(input: TokenStream) -> TokenStream {
 
     let ast = syn::parse_macro_input(&input).expect("Couldn't parse item");
 
-    let result = builder_for_struct(ast);
+    let result = builder_for_struct(ast).to_string();
 
-    format!("{}", result).parse().expect("Couldn't parse string to tokens")
+    result.parse().expect(&format!("Couldn't parse {:?} to tokens", result))
```

Comment:
rustc and cargo usually put code or values in error messages backticks, i.e. ```"Couldn't parse `{:?}` to tokens"```. (Changing here and everywhere else this will require you to adjust a bunch of tests though)